bool close
float32 time_stamp
